zebrad/src/components/inbound.rs

```diff
@@ -320,16 +320,13 @@ impl Service<zn::Request> for Inbound {
                     .map_ok(zn::Response::Blocks)
                     .boxed()
             }
-            zn::Request::TransactionsById(_transactions) => {
-                // `zcashd` returns a list of found transactions, followed by a
-                // `NotFound` message if any transactions are missing. `zcashd`
-                // says that Simplified Payment Verification (SPV) clients rely on
-                // this behaviour - are there any of them on the Zcash network?
-                // https://github.com/zcash/zcash/blob/e7b425298f6d9a54810cb7183f00be547e4d9415/src/main.cpp#L5632
-                // We'll implement this request once we have a mempool:
-                // https://en.bitcoin.it/wiki/Protocol_documentation#getdata
-                debug!("ignoring unimplemented request");
-                async { Ok(zn::Response::Nil) }.boxed()
+            zn::Request::TransactionsById(transactions) => {
+                self.mempool.clone().oneshot(mempool::Request::TransactionsById(transactions)).map_ok(|resp| match resp {
+                    mempool::Response::Transactions(transactions) => zn::Response::Transactions(transactions),
+                    _ => unreachable!("Mempool componenet should always respond to a `TransactionsById` request with a `Transactions` response"),
```

Comment:
We can break this out a bit for readability:

```suggestion
                let request = mempool::Request::TransactionsById(transactions);
                self.mempool.clone().oneshot(request).map_ok(|resp| match resp {
                    mempool::Response::Transactions(transactions) => zn::Response::Transactions(transactions),
                    _ => unreachable!("Mempool componenet should always respond to a `TransactionsById` request with a `Transactions` response"),
```